data/data-pipeline/data_pipeline/content/config/generate_data_dictionary.py

```diff
@@ -0,0 +1,185 @@
+
+PERCENTILE_EXPLANATION = (
+    "All percentiles are floored (rounded down to the nearest percentile). "
+    + "For example, 89.7th percentile is rounded down to 89 for this field."
+)
+LOW_PERCENTILE_EXPLANATION = "This percentile is reversed, meaning the lowest raw numbers become the highest percentiles."
+ISLAND_AREAS_EXPLANATION = (
+    "Because not all data is available for the Nation, Puerto Rico, "
+    + "and the Island Territories, this uses different underlying data for the islands."
+)
+
+
+OUTPUT_PATH = (
+    settings.APP_ROOT / "data" / "score" / "downloadable" / "codebook.csv"
+)
+
+
+def _set_up_infrastructure(
+    yaml_path: pathlib.Path,
+    fields_to_store: list,
+):
+    """helper function to parse yaml"""
+    yaml_contents = yaml.safe_load(
+        open(yaml_path, "r", encoding="utf-8").read()
+    )
+    if "sheets" in yaml_contents:
+        fields = []
+        for sheet in yaml_contents["sheets"]:
+            # excel yaml has ability for many sheets
+            fields.extend([single_field for single_field in sheet["fields"]])
+    else:
+        # other files do not
+        fields = yaml_contents["fields"]
+
+    to_frame_dict = {CEJST_SCORE_COLUMN_NAME: []} | {
+        field: [] for field, _ in fields_to_store
+    }
+
+    return to_frame_dict, fields
+
+
+def _create_df_from_yaml(
+    yaml_path: pathlib.Path,
+    fields_to_store: list,
+    native_score_column: str = CEJST_SCORE_COLUMN_NAME,
+):
+    """helper function to create a dataframe from yaml"""
+    to_frame_dict, fields = _set_up_infrastructure(
+        yaml_path=yaml_path,
+        fields_to_store=fields_to_store,
+    )
+    for details in fields:
+        to_frame_dict[CEJST_SCORE_COLUMN_NAME].append(
+            details[native_score_column]
+        )
+        for column_name, dict_field in fields_to_store:
+            try:
+                to_frame_dict[column_name].append(details[dict_field])
+            except KeyError:
+                assert column_name != CEJST_SCORE_COLUMN_NAME
+                to_frame_dict[column_name].append(np.nan)
+    return pd.DataFrame(to_frame_dict).set_index(CEJST_SCORE_COLUMN_NAME)
+
+
+def _get_datatype(
+    input_name: str,
+    input_type: str,
+    percentile_string: str = PERCENTILE_STRING,
+    loss_rate_string: str = LOSS_RATE_STRING,
+):
+    """helper to convert datatype"""
+    if percentile_string in input_name:
+        return "percentile"
+    elif loss_rate_string in input_name:
+        return "rate"
+    else:
+        return input_type
+
+
+def _get_calculation_notes(column_name):
+    """produces calculation notes"""
+    calculation_notes = []
+    if PERCENTILE_STRING in column_name:
+        calculation_notes += [PERCENTILE_EXPLANATION]
+    if LOW_STRING in column_name:
+        calculation_notes += [LOW_PERCENTILE_EXPLANATION]
+    if ISLAND_STRING in column_name:
+        calculation_notes += [ISLAND_AREAS_EXPLANATION]
+    return " ".join(calculation_notes)
```

Comment:
We should add these as types on the YAML config rather than deducing from the column name?

Reply:
Jorge's suggestion makes a lot of sense to me! Could put booleans on the YAML config, like, 

```
is_low_field: True
is_island_areas_field: True
```

etc

Reply:
I was hoping that our new field names after the refactor would have programmatic column names that would make this clear, but I'm happy to put this in the yaml!

Reply:
I think we can punt to the refactor, in the interest of merging this PR.